scipy/stats/tests/test_multivariate.py

```diff
@@ -1226,6 +1243,71 @@ def test_fit_fix_cov(self):
                                                      ).sum()
         assert logp_perturbed < logp_fix
 
+    @pytest.mark.parametrize('X_ndim', [3])
+    @pytest.mark.parametrize('dimensions', [[1], 
+                                            [-1, 1]])
+    @pytest.mark.parametrize('frozen', [True, False])
+    @pytest.mark.parametrize('cov_object', [True, False])
+    def test_marginal_distribution(self, X_ndim, dimensions, frozen, cov_object):
+        rng = np.random.default_rng(413911473)
+        mean = rng.standard_normal(X_ndim)
+        A = rng.standard_normal((X_ndim, X_ndim))
+        cov = A @ A.T
+        
+        if cov_object:
+            cov = _covariance.CovViaPrecision(cov)
+
+        # number of points at which to evaluate marginal PDF
+        x = np.random.standard_normal((4, len(dimensions)))
+        X = multivariate_normal(mean, cov)
+        
+        if frozen:
+            Y = X.marginal(dimensions)
+            res = Y.pdf(x)
+        else: 
+            Y = multivariate_normal.marginal(dimensions, mean, cov)
+            res = Y.pdf(x)
+
+        ref = marginal_pdf(X, X_ndim, dimensions, x)
+        assert_allclose(ref, res)
+
+    def test_marginal_input_validation(self):
+        rng = np.random.default_rng(413911473)
+        mean = rng.standard_normal(3)
+        A = rng.standard_normal((3, 3))
+        cov = np.dot(A, A.transpose())        
+
+        X = multivariate_normal(mean, cov)
+
+        msg = r"Dimensions \[3\] are invalid .*"
+        with pytest.raises(ValueError, match=msg):
+            X.marginal(3)
+
+        with pytest.raises(ValueError, match=msg):
+            X.marginal([0, 1, 2, 3])
+
+        msg = r"All elements of `dimensions` must be unique."
+        with pytest.raises(ValueError, match=msg):
+            X.marginal([2, -1])
+
+        with pytest.raises(ValueError, match=msg):
+            X.marginal([[0, 1]])
+        
+        msg = r"Elements of `dimensions` must be integers."
+        with pytest.raises(ValueError, match=msg):
+            X.marginal([1.1, 2.0])
+    
+    def test_special_cases(self):
```

Comment:
It'd help to make this test name a bit more specific

Reply:
`test_multivariate_special_cases`?

Reply:
In the file `test_multivariate` and in a class `TestMultivariateNormal`, I am not sure that would add much information. I could see `test_marginal_special_cases`, since it's right after `test_marginal_distribution` and `test_marginal_input_validation`.